### ISSUE TITLE
Fine-tune in anomaly detection

### ISSUE THREAD

**Issue by zuijiayule** [moment-timeseries-foundation-model/moment#62]
I have some questions about anomaly detection
In the tutorials/anomaly_detection.ipynb tutorial, you use a segmented test set for loss calculation and forward propagation optimization, and for anomaly detection still use the labels of the true lables to compare with the predicted values to locate anomalies 

1. I don't understand what is the point of this prediction since the raw data has true labels to compare? What should I do if I want to detect the dataset without true labels directly by the trained model at the time of detection?
2. I tried fine tuning with a test set using the code in your tutorials/anomaly_detection.ipynb. Then extracted the threshold when the F1 score is maxed out. Using that threshold for detection data, I used the training set for detection but it did not work well. The training set has no anomalies in the truth labels but the model detects many anomalies.

Translated with DeepL.com (free version)

**Comment by PoleGeogry** [comment]
@zuijiayule 
Sorry to bother you, is your problem solved? I also want to fine-tune the model on my dataset and follow its tutorial to reproduce the failure of basic zero-sample anomaly detection and the failure of fine-tuning the model. Could you please share your method? Thank you very much

**Comment by zuijiayule** [comment]
dear @PoleGeogry, 
I will share my attempt and hope it will be helpful to you.
1. First of all, thank the author. This is an exciting research. But in fact, I did not achieve good results with it. The author used the F1 adjustment function, which beautified the results to a certain extent. You can take a detailed look at the principle of the F1 adjustment function, which uses the true label to correct the detection results.
2. In terms of anomaly detection, the Moment model is essentially, as my understanding, a reconstruction-based anomaly detection method. This makes this type of model sensitive to anomalies with higher amplitudes, but powerless to other anomalies.
3. I suggest that you do not use the research script, it is very large and I believe there are some bugs that have not been fixed. But you can learn some of its engineering methods, combined with the simplified version of moment_FM tutorial, to reproduce the function yourself.
4. Other classic models are always more stable, but I still thank the author for his contribution, which give me a lot of engineering ideas from it.

Hope it helps you

**Comment by PoleGeogry** [comment]
@zuijiayule 
Thank you for sharing, which is very helpful for a novice like me who is just getting started with time series anomaly detection. Thank you again for your guidance
